### PR TITLE
Use RELEASE_ASSERT instead of raise(SIGABRT) in kill_request filter

### PR DESCRIPTION
**Pull request by qqustc**
Commit Message: Use `RELEASE_ASSERT` instead of `raise(SIGABRT)` in kill_request filter to provide better logging when the filter is crashing Envoy.
Additional Description: N/A
Risk Level: low
Testing: unit test, integration test
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

/assign @antoniovicente

### CODE REVIEW

**Review by antoniovicente — COMMENTED** [review]
Looks good.  I don't know what the format error is about.

File: source/extensions/filters/http/kill_request/kill_request_filter.cc [hunk 1]
```diff
@@ -81,7 +81,7 @@ Http::FilterHeadersStatus KillRequestFilter::encodeHeaders(Http::ResponseHeaderM
 
   if (isKillRequest(headers) && isKillRequestEnabled()) {
     // Crash Envoy.
-    raise(SIGABRT);
+    RELEASE_ASSERT(false, "KillRequestFilter is crashing Envoy!!!");
```
[the remark this answers is not in the document]
Comment (yanavlasov) [reply source/extensions/filters/http/kill_request/kill_request_filter.cc:84]: I think these death tests are already there in the kill_request_filter_test.cc

[the remark this answers is not in the document]
Comment (antoniovicente) [reply source/extensions/filters/http/kill_request/kill_request_filter.cc:84]: Right.  My request is to change expectations like:
`EXPECT_DEATH(filter_->decodeHeaders(request_headers_, false), "");`
to:
`EXPECT_DEATH(filter_->decodeHeaders(request_headers_, false), "KillRequestFilter is crashing Envoy!!!");`

[the remark this answers is not in the document]
Comment (qqustc) [reply source/extensions/filters/http/kill_request/kill_request_filter.cc:84]: thanks. Updated the test.

**Pull request merged**